.github/workflows/linux-ci.yml

```diff
@@ -17,17 +17,17 @@ jobs:
       matrix:
         # Github Actions requires a single row to be added to the build matrix.
         # See https://help.github.com/en/articles/workflow-syntax-for-github-actions.
-        name: [ubuntu-18.04-gcc-9, ubuntu-18.04-clang-10]
+        name: [ubuntu-20.04-gcc-9, ubuntu-20.04-clang-10]
```

Comment:
18.04 has Boost 1.65 and not the minimum 1.67 that GTSAM supports. As such, that serialization bug that we had about deepcopies rears its head again with 18.04.

We'll have to do a custom install of Boost if you want 18.04 which will significantly increase CI time. Thoughts?

Reply:
Help, we lost 18.04 for GTSAM???

Reply:
Nope. We do a custom Boost install for 1.67 in GTSAM. That adds a good 5 minutes to the CI.

Reply:
Tell me again why we do not support the default boost version on 1804?

Reply:
Because Boost versions before 1.67 have a bug in their serialization where only the object in question is serialized and not its constituent objects. The best analogy is that serialization would do a shallow copy when what we really need is a deep copy.

Reply:
I think it's before 1.65 : https://github.com/borglab/gtsam/blob/6f9561eb04c0d930abfdb39e2a18c4f4d5483e3e/INSTALL.md

Reply:
That doc needs to be updated. If you see the CI for GTSAM, we install 1.67

Reply:
Yes, but I think it works from 1.65. I can make a VM and test.

Reply:
@ProfFan any updates on this?

Reply:
FYI https://github.com/borglab/gtsam/issues/590#issuecomment-761906003

Reply:
Okay, I have a resolution for this: I am just setting Ubuntu 20.04 for the clang CI. That way we are still supporting Boost 1.65 (at least for GCC).

Reply:
Ping @ProfFan

Reply:
I think this is good, but